### DISCUSSION TITLE
Does a cross-platform "in view" component exist?

### DISCUSSION THREAD

**Discussion by jonsherrard** [nandorojo/solito#238] [General]
I'd like to run some animations when a component comes into view, on mobile and web.

There are a couple of solutions around:

* https://www.framer.com/docs/use-in-view/ is my favourite on the web
* https://github.com/wellyshen/react-cool-inview is great too

https://www.npmjs.com/package/@svanboxel/visibility-sensor-react-native - seems to be the most recently update on native

Does anyone know if a cross-platform solito-friendly library exists/

Or should I write an abstraction that uses one library or the other on each platform?

**Comment by nandorojo** [comment]
native doesn’t have an IntersectionObserver API annoyingly. your best best is to use FlatList/FlashList which have a callback for viewable items changing, or finding some native-based solution
